packages/admin-ui/src/table/resolvers/menu.tsx

```diff
@@ -0,0 +1,75 @@
+import type { ReactNode } from 'react'
+import React from 'react'
+import invariant from 'tiny-invariant'
+
+import { Skeleton } from '../../components/Skeleton'
+import { Menu, MenuButton, MenuItem, useMenuState } from '../../menu'
+import type { ResolverRenderProps } from './resolver-core'
+import { createResolver, defaultRender } from './resolver-core'
+
+export function menuResolver<T>() {
+  return createResolver<T, 'menu', MenuResolver<T>>({
+    cell: function MenuResolver({ item, column, context }) {
+      if (context.status === 'loading') {
+        return <Skeleton csx={{ size: 24 }} />
+      }
+
+      const { resolver } = column
+
+      invariant(
+        resolver,
+        'Resolver prop is required while using the menu resolver'
+      )
+
+      const { actions } = resolver
+
+      const state = useMenuState()
+
+      const data = (
+        <>
+          <MenuButton
+            state={state}
+            onClick={(e) => e.stopPropagation()}
+            variant="neutralTertiary"
+            labelHidden
+          />
+          <Menu state={state}>
+            {actions.map((action, index) => {
+              return (
+                <MenuItem
+                  label={action.label}
+                  onClick={(e) => {
+                    e.stopPropagation()
+                    action.onClick(item)
+                  }}
+                  disabled={action?.disabled}
+                  critical={action?.critical}
+                  icon={action?.icon}
+                  key={`column-${String(column.id)}-menu-item-${index}`}
```

Comment:
```suggestion
                  key={index}
```
Keys only need to be unique among the same collection. Index should be fine since you don't reorder or filter.